### ISSUE TITLE
Docs: Configuration of Gradle when using logging and ShadowJar plugin

### ISSUE THREAD

**Issue by roamingthings** [aws-powertools/powertools-lambda-java#1397]
### What were you searching in the docs?

I have been using the logging component of Powertools together with the `com.github.johnrengelman.shadow` plugin for Gradle. I've also configured log4j2 to use the provided JSON format.

When running the Lambda Function I got errors from the logging system about the output pattern (many lines like `ERROR StatusLogger Unrecognized format specifier [d]`). The strange thing has been that the errors related to a pattern that I didn't configure.

I have searching the documentation mentioning something like that.

### Is this related to an existing documentation section?

https://docs.powertools.aws.dev/lambda/java/core/logging/#initialization

### How can we improve?

When using the ShadowJar plugin and configuring the logging component as described in https://docs.powertools.aws.dev/lambda/java/core/logging/#initialization you also have to configure a transformer for the ShadowJar plugin:

```
tasks.withType<ShadowJar>() {
    transform(Log4j2PluginsCacheFileTransformer::class.java)
}
```

Unfortunately this solution is not easy to find since the error is not pointing in that direction.

### Got a suggestion in mind?

Not yet. I wanted to first document this finding in case someone else is having this issue.

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

**Comment by scottgerring** [comment]
Hey @roamingthings thanks for raising this!
I believe we have a bit of a gap in our gradle documentation and setup instructions. We've started to address this, recently [added a gradle example](https://github.com/aws-powertools/powertools-lambda-java/tree/main/examples/powertools-examples-core/gradle) and [fixed up the SAM template for Java 17](https://github.com/aws/aws-sam-cli-app-templates/pull/444). 

If you have an idea of how we could document this that'd be great, and certainly it's something we should add to our gradle example.

**Comment by roamingthings** [comment]
I came across this when combining Powertools with the default Micronaut 4 project template.

**Comment by roamingthings** [comment]
Adding it to the Gradle example would be a good start. However, at least I have only been looking at the documentation. Maybe it fits the [Install](https://docs.powertools.aws.dev/lambda/java/core/logging/#install) section?

**Comment by scottgerring** [comment]
@jeromevdl / @mriccia what do you think? You mentioned how shade works is a function of deployment - e.g. SAM doesn't use it. Not sure what we should do here.

**Comment by eldimi** [comment]
> @jeromevdl / @mriccia what do you think? You mentioned how shade works is a function of deployment - e.g. SAM doesn't use it. Not sure what we should do here.

The [AWS Lambda documentation](https://docs.aws.amazon.com/lambda/latest/dg/java-package.html) instructs to create a shaded jar for maven and .zip file for gradle. I am curious why this is so.
I have used shaded jar built with gradle to deploy with SAM, skipping the sam build step.
Spring Cloud Function documentation also instructs to created a shaded jar with gradle to deploy an AWS Lambda function

**Comment by roamingthings** [comment]
This text found on [Best practices for working with AWS Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html) confuses me even more

> Reduce the time it takes Lambda to unpack deployment packages authored in Java by putting your dependency .jar files in a separate /lib directory. This is faster than putting all your function’s code in a single jar with a large number of .class files. See [Deploy Java Lambda functions with .zip or JAR file archives](https://docs.aws.amazon.com/lambda/latest/dg/java-package.html) for instructions.

**Comment by jeromevdl** [comment]
I don't remember where I saw that but shaded jar are actually faster than zip with lib folder. 
But there is a problem with Multi-Release Jars when using shading: https://awstip.com/packaging-deploying-java11-based-lambda-functions-19c456e062a9.

Anyway, not sure what was the initial question? Our documentation does not specify how to actually build a package. Is it not enough to reference the official AWS doc? If we start to document maven, gradle, using SAM or not, using shade or assembly or whatnot, plus the differences between java 8 and others (because of aspectj), we'll have tens of tabs in the documentation. We already have 4 :( 

Also, I think we discussed that already, but should we remove the `maven-shade-plugin` configuration from our SAM examples, as SAM is not using it and it can be misleading?
